### PR TITLE
Add news-details and dialog window

### PR DESCRIPTION
**Pull request by yaroslavbilokin**

### CODE REVIEW

File: src/components/dialog-window/dialog-window.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  DialogTitle,
+  Typography
+} from '@material-ui/core/';
+import { useDispatch, useSelector } from 'react-redux';
+import { config } from '../../configs';
+import { setDialogStatus } from '../../redux/dialog-window/dialog-window.actions';
+import { StandardButton } from '../buttons';
+
+const { CANCEL_TITLE } = config.buttonTitles;
+const { ACCEPT_BUTTON_STYLE } = config.buttonStyles;
```
Comment (VolodymyrTrach) [inline src/components/dialog-window/dialog-window.js:15]: why do you use config for styles instead of button.style.js?

File: src/utils/client.js [hunk 2]
```diff
@@ -103,8 +103,8 @@ const createNewsItem = async (news) => {
   client.resetStore();
 };
 
-const updateNewsItem = (id, news) => {
-  client.mutate({
+const updateNewsItem = async (id, news) => {
```
Comment (Alan18081) [inline src/utils/client.js:106]: First of all, move this code to another file, in `news` folder, call it something like `news.operations.js`. File `client` should only configure client and export it.

Comment (Alan18081) [reply src/utils/client.js:106]: Also, please, return the result of the client's call from this function to get updated fields back

File: src/redux/news/news.reducer.js [hunk 3]
```diff
@@ -1,7 +1,60 @@
...
+  SET_NEWS_ITEM
+} from './news.types';
+
+import { config } from '../../configs';
+
+const { languages } = config;
 
 const initialState = {
   list: [],
+  newsItem: {
```
Comment (Alan18081) [inline src/redux/news/news.reducer.js:14]: It should have an initial value of `null`

Comment (Alan18081) [reply src/redux/news/news.reducer.js:14]: Also, please, rename field from `newsItem` to `activeArticle`

File: src/pages/news/news-page/news-page.js [hunk 4]
```diff
@@ -19,17 +30,30 @@ const tableTitles = config.tableHeadRowTitles.news;
...
+  }));
   const dispatch = useDispatch();
 
   useEffect(() => {
     dispatch(getNews());
   }, [dispatch]);
 
-  const loading = useSelector(({ News }) => News.loading);
+  const openSuccessSnackbar = (eventHandler) => {
+    dispatch(setDialogTitle(REMOVE_TITLE));
```
Comment (Alan18081) [inline src/pages/news/news-page/news-page.js:44]: Please, try to reduce this long list of action creators to two:
- `showDialog`that will take object with all of the data for the dialog
- `closeDialog` that will reset all of the data and close the dialog

File: src/redux/news/news.sagas.js [hunk 5]
```diff
@@ -23,13 +35,27 @@ function* handleNewsLoad() {
   }
 }
 
+function* handleNewsItemLoad({ payload }) {
+  try {
+    yield put(showLoader());
```
Comment (Alan18081) [inline src/redux/news/news.sagas.js:40]: Simplify action creators `showLoader` and `hideLoader` to `setLoading` that will take `true` or `false` as arguments

File: src/redux/news/news.sagas.js [hunk 6]
```diff
@@ -23,13 +35,27 @@ function* handleNewsLoad() {
...
+    const newsItem = yield call(getNewsItemById, payload);
+    yield put(setNewsItem(newsItem.data.getNewsById));
+    yield put(hideLoader());
+  } catch (error) {
+    console.log(error);
```
Comment (Alan18081) [inline src/redux/news/news.sagas.js:45]: Add error handling to show some notification message when an error happens

File: src/redux/news/news.sagas.js [hunk 7]
```diff
@@ -23,13 +35,27 @@ function* handleNewsLoad() {
...
     yield call(createNewsItem, payload);
-    yield put(push('/'));
+    const news = yield call(getAllNews, null);
+    yield put(setNews(news.data.getAllNews));
     yield put(setSnackBarSeverity('success'));
     yield put(setSnackBarMessage(SUCCESS_ADD_STATUS));
     yield put(setSnackBarStatus(true));
+    yield put(push('/'));
   } catch (err) {
     console.log(err);
```
Comment (Alan18081) [inline src/redux/news/news.sagas.js:60]: Add error handling to show some notification message when an error happens

File: src/redux/news/news.sagas.js [hunk 8]
```diff
@@ -47,8 +73,23 @@ function* handleNewsDelete({ payload }) {
...
+function* handleNewsUpdate({ payload }) {
+  const { id, newNewsItem } = payload;
+  try {
+    yield put(showLoader());
+    yield call(updateNewsItem, id, newNewsItem);
+    const news = yield call(getAllNews, null);
+    yield put(setNews(news.data.getAllNews));
+    yield put(push('/'));
+  } catch (error) {
+    console.log(error);
```
Comment (Alan18081) [inline src/redux/news/news.sagas.js:85]: Add error handling to show some notification message when an error happens

File: src/pages/news/news-page/news-page.js [hunk 9]
```diff
@@ -19,17 +30,30 @@ const tableTitles = config.tableHeadRowTitles.news;
...
+    dispatch(setButtonTitle(REMOVE_TITLE));
+    dispatch(setEventHandler(eventHandler));
+    dispatch(setDialogStatus(true));
+  };
 
   const newsDeleteHandler = (id) => {
-    dispatch(deleteNewsItem(id));
+    const removeNews = async () => {
+      dispatch(setDialogStatus(false));
+      await dispatch(deleteNewsItem(id));
```
[the remark this answers is not in the document]
Comment (yaroslavbilokin) [reply src/pages/news/news-page/news-page.js:54]: Fixed!

File: src/utils/useNewsHandlers.js [hunk 10]
```diff
@@ -0,0 +1,38 @@
+import { useState } from 'react';
```
[the remark this answers is not in the document]
Comment (yaroslavbilokin) [reply src/utils/useNewsHandlers.js:1]: Renamed!

File: src/redux/dialog-window/dialog-window.reducer.js [hunk 11]
```diff
@@ -0,0 +1,46 @@
+const initialState = {
+  dialogStatus: false,
+  dialogTitle: '',
+  dialogContent: '',
+  buttonTitle: '',
+  eventHandler: () => {}
+};
+
+const dialogWindowState = (state = initialState, action = {}) => {
```
[the remark this answers is not in the document]
Comment (yaroslavbilokin) [reply src/redux/dialog-window/dialog-window.reducer.js:17]: Fixed!

File: src/redux/news/news.sagas.js (hunk 5 above)
Comment (yaroslavbilokin) [reply src/redux/news/news.sagas.js:40]: Fixed!

File: src/redux/dialog-window/dialog-window.reducer.js [hunk 12]
```diff
@@ -1,46 +1,30 @@
+import { CLOSE_DIALOG, SHOW_DIALOG } from './dialog-window.types';
+
 const initialState = {
   dialogStatus: false,
```
[the remark this answers is not in the document]
Comment (yaroslavbilokin) [reply src/redux/dialog-window/dialog-window.reducer.js:4]: You're right! Renamed!

**Pull request merged**